programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -325,6 +326,14 @@ pub fn create_program_runtime_environment_v1<'a>(
         SyscallBigModExp::call,
     )?;
 
+    // Poseidon
+    register_feature_gated_function!(
+        result,
+        enable_poseidon_bn254_syscall,
+        b"sol_poseidon_bn254",
+        SyscallPoseidonBn254::call,
```

Comment:
What do you think about just defining a `SyscallPoseidon` syscall that take in some sort of `curve_id` or variant identifier as a parameter? maybe something like [this](https://github.com/solana-labs/solana/blob/master/programs/bpf_loader/src/syscalls/mod.rs#L926). For now, we can just have the `bn254` variant, but down the line, we can add Poseidon optimized for bls family of curves and others.

Reply:
Good idea. I think it would also make sense to make endianness configurable this way - for now I just assumed that the inputs and outputs to be big endian, which is what Light Protocol expects.